packages/mongo-nav/src/on-element-click-provider/OnElementClickProvider.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { createContext, useContext } from 'react';
+import { NavElement } from '../types';
+
+type OnElementClickType = (
+  type: NavElement,
+  additionalWork?: (e: React.MouseEvent) => void,
+) => (e: React.MouseEvent) => void;
+
+interface OnElementClickProviderProps {
+  children: React.ReactNode;
+  onElementClick: (type: NavElement, e: React.MouseEvent) => void;
+}
+
+const OnElementClickContext = createContext<OnElementClickType>(
+  (_navElement: NavElement, additionalWork?: (e: React.MouseEvent) => void) => (
+    e: React.MouseEvent,
+  ) => {
+    return additionalWork?.(e);
+  },
+);
+
+export function useOnElementClick() {
+  return useContext(OnElementClickContext);
+}
+
+export default function OnElementClickProvider({
+  children,
+  onElementClick,
+}: OnElementClickProviderProps) {
+  const wrappedOnElementClick = (
```

Comment:
Need this memoized. Your usage of `useMemo` below requires the second argument: `useMemo(functionToMemo, dependencyArrayLikeUseEffect)`.

So for example `useMemo(wrappedOnElementClick, [onElementClick])`

Reply:
👍